### PR TITLE
CLDR-15187 Encapsulate locale name constants for mis, mul, root, und, zxx

### PR DESCRIPTION
**Pull request by btangmu**
-New class LocaleNames has constants MIS, MUL, ROOT, UND, ZXX

CLDR-15187

- [x] This PR completes the ticket.

### CODE REVIEW

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/AddUser.java [hunk 1]
```diff
@@ -123,7 +124,7 @@ private AddUserResponse doAddNewUser(AddUserRequest request, CookieSession sessi
     private String getNewUserLocales(String requestLocales) {
         requestLocales = LocaleNormalizer.normalizeQuietly(requestLocales);
         if (requestLocales.isEmpty()) {
-            return "mul";
+            return LocaleNames.MUL;
```
Comment (srl295) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/AddUser.java:127]: This is a bug I'm working on separately. Can you remove it from this ticket?

**Review by srl295 — APPROVED** [review]

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/AddUser.java (hunk 1 above)
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/AddUser.java:127]: I guess its OK for now

mul is also used for the sandbox (which is the meaning here) . That meaning should be ideally encapsulated separately

Comment (btangmu) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/AddUser.java:127]: maybe this is at least a step in that direction? Would they both still be named "mul"?

**Review by macchiati — APPROVED** [review]

**Comment by btangmu** [comment]
This PR had conflicts with main earlier today, for TestPersonNameFormatter.java

At this moment "This branch has no conflicts with the base branch", following https://github.com/unicode-org/cldr/pull/2554

Nevertheless I don't want to create difficulties for the work-in-progress on person names so I'll postpone merging this -- and might step back and do this in stages, postponing changes to files that have work-in-progress for other tickets...

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/AddUser.java (hunk 1 above)
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/AddUser.java:127]: yes, they are both named mul

**Comment by srl295** [comment]
> This PR had conflicts with main earlier today, for TestPersonNameFormatter.java
> 
> At this moment "This branch has no conflicts with the base branch", following #2554
> 
> Nevertheless I don't want to create difficulties for the work-in-progress on person names so I'll postpone merging this -- and might step back and do this in stages, postponing changes to files that have work-in-progress for other tickets...

actually if you rebase it it should be fine to merge

**Review by srl295 — APPROVED** [review]

**Pull request closed without merging**